src/distribution.cpp

```diff
@@ -0,0 +1,312 @@
+#include "openmc/distribution.h"
+
+#include <algorithm> // for copy
+#include <cmath>     // for sqrt, floor, max
+#include <iterator>  // for back_inserter
+#include <numeric>   // for accumulate
+#include <stdexcept> // for runtime_error
+#include <string>    // for string, stod
+
+#include "openmc/error.h"
+#include "openmc/math_functions.h"
+#include "openmc/random_lcg.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+//==============================================================================
+// Discrete implementation
+//==============================================================================
+
+Discrete::Discrete(pugi::xml_node node)
+{
+  auto params = get_node_array<double>(node, "parameters");
+
+  std::size_t n = params.size();
+  std::copy(params.begin(), params.begin() + n/2, std::back_inserter(x_));
+  std::copy(params.begin() + n/2, params.end(), std::back_inserter(p_));
+
+  normalize();
+}
+
+Discrete::Discrete(const double* x, const double* p, int n)
+  : x_{x, x+n}, p_{p, p+n}
+{
+  normalize();
+}
+
+double Discrete::sample() const
+{
+  int n = x_.size();
+  if (n > 1) {
+    double xi = prn();
+    double c = 0.0;
+    for (int i = 0; i < n; ++i) {
+      c += p_[i];
+      if (xi < c) return x_[i];
+    }
+    throw std::runtime_error{"Error when sampling probability mass function."};
+  } else {
+    return x_[0];
+  }
+}
+
+void Discrete::normalize()
+{
+  // Renormalize density function so that it sums to unity
+  double norm = std::accumulate(p_.begin(), p_.end(), 0.0);
+  for (auto& p_i : p_)
```

Comment:
Nit: The established pattern in these files is to use { } even if only a single statement.